packages/playwright-core/src/server/frames.ts

```diff
@@ -1399,6 +1418,33 @@ export class Frame extends SdkObject {
       return injectedScript.extend(source, arg);
     }, { source, arg });
   }
+
+  private async _resolveFrame(progress: Progress, selector: string, options: types.StrictOptions & types.TimeoutOptions): Promise<{ frame: Frame, element: ElementHandle<Element>, info: SelectorInfo }> {
+    const elementPath: ElementHandle<Element>[] = [];
+    progress.cleanupWhenAborted(() => {
+      // Do not await here to avoid being blocked, either by stalled
+      // page (e.g. alert) or unresolved navigation in Chromium.
+      for (const element of elementPath)
+        element.dispose();
+    });
+
+    let frame: Frame = this;
+    const frameChunks = splitSelectorByFrame(selector);
+    let info: SelectorInfo;
+
+    let element: ElementHandle<Element>;
+    for (let i = 0; i < frameChunks.length && progress.isRunning(); ++i) {
+      info = this._page.parseSelector(frameChunks[i], options);
+      const task = dom.waitForSelectorTask(info, 'attached');
+      const handle = await frame._scheduleRerunnableHandleTask(progress, info.world, task);
+      element = handle.asElement() as dom.ElementHandle<Element>;
+      if (i < frameChunks.length - 1) {
+        frame = (await element.contentFrame())!;
```

Comment:
if (!frame) throw new Error(`Selector "..." resolved to <preview> instead of an iframe`);

Reply:
nit: drop the exclamation mark here